Make StringContextTest pass on Windows

linasm opened the pull request:
One assertion in StringContextTest."f interpolator baseline"() was relying on platform specific newline encoding, making it fail when run on Windows.

som-snytt commented on test/junit/scala/StringContextTest.scala:
```diff
@@ -224,11 +224,7 @@ class StringContextTest {
-          |------
-          |matter
-          |""".stripMargin,
+      f" mind%n------%nmatter%n" -> String.format(" mind%n------%nmatter%n"),
```
You can preserve the visual pun by

```
      f" mind%n------%nmatter%n".lines.mkString("\n") ->
       """| mind
          |------
          |matter
          |""".stripMargin.lines.mkString("\n"),

```

`lines.mkString` is an easy EOL normalizer.

linasm replied:
I thought about something similar but then I realized that this would somewhat weaken the test - i.e. when run on Windows, it would not assert that "%n" gets converted to "\r\n", as it should be. So I decided to only adjust the "expected" side of the assertion.

som-snytt replied:
Right. How about

```
      f" mind%n------%nmatter" ->
       """| mind
          |------
          |matter
          |""".stripMargin.lines.mkString(compat.Platform.EOL),

```

linasm commented:
Sorry, forgot to squash the commits. Should I submit a new Pull Request?

SethTisue commented:
you'll need to squash before this can be merged, but you don't need to make a whole new PR, you can force-push to your existing branch.

linasm commented:
Yep, squashed the extra commit and force pushed in my last attempt, hopefully the end result is fine.

mpociecha commented:
By the way, such a little hint for the future: `git pull --rebase` is really awesome command and I highly recommend to use it. :)

SethTisue commented:
the PR is still showing 9 commits. perhaps you pushed to the wrong branch, or something?

Commit:
Make StringContextTest pass on Windows

One assertion in StringContextTest."f interpolator baseline"() was
relying on platform specific newline encoding, making it fail when run
on Windows.

linasm commented:
Commit history cleaned up.

SethTisue commented:
thanks @linasm!

btw we do use Jenkins to run the test suite on Windows on 2.11, and we plan to do the same on 2.12 before too much longer, I just haven't quite gotten to it yet. it's https://github.com/scala/scala-jenkins-infra/issues/142
